packages/account/src/predicate/predicate.ts

```diff
@@ -43,12 +43,14 @@ export class Predicate<ARGS extends InputValue[]> extends Account {
    * @param bytes - The bytes of the predicate.
    * @param provider - The provider used to interact with the blockchain.
    * @param jsonAbi - The JSON ABI of the predicate.
+   * @param predicateData - The predicate data.
    * @param configurableConstants - Optional configurable constants for the predicate.
    */
   constructor(
     bytes: BytesLike,
     provider: Provider,
     jsonAbi?: JsonAbi,
```

Comment:
@Dhaiwat10 I reckon we must make the `jsonAbi` parameter required on the Predicate constructor.

The parameter is later used on `processPredicateData` which will throw an error if `jsonAbi` is undefined.

By doing so we will also be able to rearrange the Predicate's constructor parameters order more conveniently.

Reply:
@Torres-ssf I think this could be worth handling in a separate PR, but regardless, I don't think making `jsonAbi` a required property would change the order would it? 🤔

Reply:
@Dhaiwat10 we can make the `jsonAbi` to be the second parameter if we make it a required parameter. 

TS does not allow a required function parameter to be after an optional parameter. That is why the `predicate` parameter is the second parameter.

Since we are already updating the predicate constructor parameters, I believe this should be done within this PR's scope.

Reply:
@Torres-ssf After some discussions on this PR, the constructor now accepts an object though. Do you think making the `abi` required is still needed?

Reply:
@Dhaiwat10 AFAIK, ABIs should be mandatory for any interaction _[with Contracts]_, as the SDK won't be able to decode anything (errors, logs, outputs) without them. Isn't this also the case for Scripts and Predicates?

Reply:
@arboleya That _does_ make sense but I insist on handling this change via a separate PR because there are several cases throughout our codebase where the ABI for predicates is not passed / is optional, like this one: https://github.com/FuelLabs/fuels-ts/blob/b1bcfd26a3201c2514479382ed06d6c93d853a3b/packages/fuel-gauge/src/predicate/predicate-evaluations.test.ts#L38

I want to carefully understand why ABIs for predicates were kept optional in the first place.

Reply:
That's alright, but please create a follow-up issue so we don't forget it. 🙂

Reply:
#1843